### ISSUE TITLE
`failed to convert Descheduler minor version to float` on start

### ISSUE THREAD

**Issue by der-eismann** [kubernetes-sigs/descheduler#1419]
**What version of descheduler are you using?**

descheduler version: 0.30.0

**Does this issue reproduce with the latest release?**

yes

**Which descheduler CLI options are you using?**

`--kubeconfig container/config-staging --policy-config-file container/policy.yaml`

**Please provide a copy of your descheduler policy config file**

```yaml
apiVersion: "descheduler/v1alpha1"
kind: "DeschedulerPolicy"
nodeSelector: node.kubernetes.io/role=worker
evictLocalStoragePods: true
evictSystemCriticalPods: false
ignorePvcPods: false
strategies:
  "RemovePodsViolatingTopologySpreadConstraint":
    enabled: true
    params:
      includeSoftConstraints: true # soft = 'ScheduleAnyway' constraint
      nodeFit: true
      thresholdPriority: 200100 # only evict pods with 'default' pod priority
```

**What k8s version are you using (`kubectl version`)?**

<details><summary><code>kubectl version</code> Output</summary><br><pre>
$ kubectl version
Client Version: v1.30.1
Kustomize Version: v5.0.4-0.20230601165947-6ce0bf390ce3
Server Version: v1.30.0-eks-036c24b

</pre></details>

**What did you do?**

Start the descheduler

**What did you expect to see?**

No warning

**What did you see instead?**

This on every start:
```
W0529 16:12:11.137539  715219 descheduler.go:248] failed to convert Descheduler minor version to float
```

I don't know why the `GitVersion` is used to parse [here](https://github.com/kubernetes-sigs/descheduler/blob/release-1.30/pkg/descheduler/descheduler.go#L290-L293) - it has a format of `v20240519-v0.30.0`. According to the docs of `utilversion.ParseGeneric`:
> ParseGeneric parses a "generic" version string. The version string must consist of two or more dot-separated numeric fields (the first of which can't have leading zeroes), *followed* by arbitrary uninterpreted data (which need not be separated from the final numeric field by punctuation). For convenience, leading and trailing whitespace is ignored, and the version can be preceded by the letter "v". See also ParseSemantic.

The version it's trying to parse however is this (obtained from `descheduler version` - also see the double `v`):
```
v20240519-v0.30.0
```

My suggestion would be to either replace `VERSION?=v$(shell date +%Y%m%d)-$(shell git describe --tags)` in the [Makefile](https://github.com/kubernetes-sigs/descheduler/blob/release-1.30/Makefile#L20) with `VERSION?=$(shell git describe --tags)-$(shell date +%Y%m%d)` (but no idea what other implications that would have) or alternatively just parse `versionInfo.Minor` in the [descheduler.go](https://github.com/kubernetes-sigs/descheduler/blob/release-1.30/pkg/descheduler/descheduler.go#L290-L293). But the way it is right now is just broken.

**Comment by a7i** [comment]
I may have tagged the release incorrectly. cc: @damemi

**Comment by damemi** [comment]
I don't see a git tag for `v20240519-v0.30.0` in github, @a7i can you remember the order of the steps you did? In relation to https://github.com/kubernetes-sigs/descheduler/blob/master/docs/release-guide.md#release-process

That said we might be able to fix it by just tagging a new version. I don't know if we can delete the old k8s registry images but we might be able to

**Comment by a7i** [comment]
Yes, I followed those steps.

Looks like Descheduler version parsing broke in this commit https://github.com/kubernetes-sigs/descheduler/commit/dca2e58b8e705ad60b3c060f541e9d0a43238fe0

I can submit a fix soon

**Comment by damemi** [comment]
@a7i no worries, I wasn't implying that you didn't follow those steps. I just meant that we could trace back through that flowchart to try to figure out what could have happened :) looks like you've tracked it down though

**Comment by der-eismann** [comment]
Mega, thanks for the quick fix!

**Comment by a7i** [comment]
fixed in #1430 
/close
